### DISCUSSION TITLE
Query object becomes empty when page reloads

### DISCUSSION THREAD

**Discussion by lamuelkane** [vercel/next.js#28837] [Help]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14.17.0

### What browser are you using?

Chrome I just uodated

### What operating system are you using?

Windows

### How are you deploying your application?

next start

### Describe the Bug

When I reload the page, i loose my query params when the page reloads or when i load to that page directly passing the querys and this is not healthy because i am building an e-commerce so i worry for when a user copies a link to a particular category or product I fear the product will not load or the category will not be gotten

### Expected Behavior

I need to always get the query object when the page reloads or is loaded directly

### To Reproduce

` <Link href={{ pathname: '/shop', query: {category: 'all-products', product: `${product}`}}} >Shop</Link>  `

**Comment by florianldt** [comment]
Might be related to that discussion there: https://github.com/vercel/next.js/discussions/11484. Multiple solutions are provided.
